### ISSUE TITLE
Can´t open any file with latest build (with exiv and graphicksmagick support on Windows)

### ISSUE THREAD

**Issue by smarquespt** [luspi/photoqt-dev#28]
How to reproduce:

Open photoqt -> open file -> crash

Happens with 32 and 64bits build. 

Attach screenshot
![screenshot](https://cloud.githubusercontent.com/assets/1302668/4976108/422aaa26-68d8-11e4-9299-e9ccba6d33b7.png)

**Comment by smarquespt** [comment]
Happens again with 1.2 release. You can grab the file in http://ge.tt/9zN15j92/v/0

**Comment by luspi** [comment]
I'm sorry to hear that! Just to clarify, is the image you linked to also causing PhotoQt to crash for you? If not, would you mind sending me an image that does cause a crash (you can also send it via email: Lukas@photoqt.org) and I see if I can figure out what might be going on?

**Comment by smarquespt** [comment]
@luspi 

You can use linked image.

**Comment by luspi** [comment]
(Sorry for the delay, had some really busy days in college...)

That's odd, I can open the image here in Windows 7 without any problems...

Try installing this version: http://photoqt.org/windows/photoqt-log.exe
It is exactly the same as v1.2, except that it will create a file in you user home directory (something like C:\Users\<username>) called "photoqt-logfile" containing detailed listings and information about each step PhotoQt is doing, including any error messages. If you could either post its content (it's a simple text file) here or send it to me per email, that'd be great!

Thanks!
Lukas

**Comment by smarquespt** [comment]
@luspi 

Really strange issue. I don´t have any log file. But I notice that it happens with photos in hard drive. If a plug an android device a load photos from it´s sdcard the file is opened.

**Comment by luspi** [comment]
That really is strange. Try it without opening an image (i.e., simply start PhotoQt and close it again). This should already put something into that file...

Also: What is the exact path of an image that you're trying to open that causes PhotoQt to crash? Maybe there's some character in it that PhotoQt doesn't like (there shouldn't be, but lets make sure)...

**Comment by smarquespt** [comment]
I´ve narrow the issue. I have an user created and it´s not admin. If I install photoqt with that use account, the crash happens. But if I log in with admin accont the files are loaded fine.

**Comment by luspi** [comment]
(Also see my comment in #36)

I'm currently uploading a new installer file to the website ([link](http://photoqt.org/downloads.php?windows&32)) that properly sets all entries/shortcuts for all users no matter who is running the installer (both admin and non-admin).

See if that solves your crash, as it should remove any issue with user/admin privileges (I think, that that might have been the root of the problems you were having)...

**Comment by smarquespt** [comment]
New instaler fixed #37 but not this one. It still crash. Can you provide me a new logfile installer?

**Comment by luspi** [comment]
Actually, the new installer should have created a logfile.

But I think I finally figured out the problem:
I guess, you're user name on Windows includes some non-latin character (I'd guess an "é" from your first name)? I was able to reproduce the problem you were having by creating a user account with the fictional name "jéan" (first thing that came to my mind).
Everything seemed fine when I opened PhotoQt and from within PhotoQt opened the file, but opening PhotoQT by clicking on an image file caused troubles (it wasn't able to handle the "é", and it tried to access a QList with the faulty filename as key, however, a slightly different faulty filename was used at first => crash).

The reason for that, is, that for some reason in Windows, a filename that is passed on into the "argv" variable of the C++ main() function somehow causes troubles with proper encoding (when accessed with Qt).

However, I found a solution to that [on stackoverflow](http://stackoverflow.com/questions/4101864/windows-unicode-commandline-argv). Basically, if I get the arguments using app.arguments(), there's no problem with the encoding, it seems to work just fine.

I updated the installer on [the website](http://photoqt.org/downloads.php?windows&32), that should hopefully solve the problem for you...

**Comment by smarquespt** [comment]
You are correct. My username is "Sérgio". But even with new installer the issue is still there. Can you please try create a new user with my name and see what happens?

**Comment by luspi** [comment]
Hm, I do indeed still get a crash, too. The easiest way for me to reproduce the crash is to delete the ".photoqt" settings folder, so that it is recreated by starting photoqt. The following few times it seems to work okay.
The weird thing is, that the crash apparently is caused by GraphicsMagick, even when opening e.g. a JPEG file...

I'm currently trying to rebuild the GraphicsMagic library (there has been a newer version out since I compiled it last), hoping that that will show some improvement...

**Comment by luspi** [comment]
Okay, I compiled the new version (with some optimised adjusted configuration), and I wasn't able to reproduce the crash anymore, it seems to be working just fine now...

(Also, I added a missing dll for SVG support...)

**Comment by smarquespt** [comment]
Is there a new installer?

**Comment by smarquespt** [comment]
No. I still have the crash.

**Comment by luspi** [comment]
Okay, it seemed to work for a little while for me, but then it started crashing again, with the faulty dll being one from GraphicsMagick. I then tried to compile PhotoQt without GraphicsMagick, but still with Exiv2, and it still crashes, but this time the responsible dll is libstdc++...
I really do not understand what's going on there. However, compiling PhotoQt with both GraphicsMagick and Exiv2 disabled, seems to work.

For the time being, I think that's the best "workaround" to have at least something that works on every system ([there are installers for both](http://photoqt.org/down)). I'm currently focussing on porting PhotoQt to QML, and this comes with a very different way of handling both GraphicsMagick and Exiv2. If that still leads to the crash, then the problem must be within GraphicsMagick and/or Exiv2, and I'll report it to upstream.

**Comment by smarquespt** [comment]
@luspi 

Basic setup file works fine.

**Comment by luspi** [comment]
PhotoQt is moving away from GitHub to GitLab ([more info](http://photoqt.org/news/1434375887)). This issue is duplicated here: https://gitlab.com/luspi/photoqt-legacy/issues
